src/ota_server.c

```diff
@@ -0,0 +1,346 @@
+/*
+Copyright © 2021 Ci4Rail GmbH
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include <sys/param.h>
+#include "esp_log.h"
+#include "lwip/err.h"
+#include "lwip/sockets.h"
+#include "esp_event.h"
+#if CONFIG_OTA_CDC_ECM
+#include "tusb_cdc_ecm.h"
+#include "tinyusb.h"
+#elif CONFIG_OTA_ETHERNET
+#include "driver/gpio.h"
+#include "esp_netif.h"
+#include "esp_eth.h"
+#include "lwip/ip.h"
+#endif
+#include "send_eeprom.h"
+#include "ota_update.h"
+#include <errno.h>
+
+static const char *TAG = "ota-server";
+
+#define OTA_SERVER_PORT 1234
+#define OTA_THREAD_STACK_SIZE 8192
+
+#if CONFIG_OTA_CDC_ECM
+/* configure cdc-ecm driver to use dhcp */
+static const ip_addr_t ipaddr  = IPADDR4_INIT_BYTES(0, 0, 0, 0);
+static const ip_addr_t netmask = IPADDR4_INIT_BYTES(0, 0, 0, 0);
+static const ip_addr_t gateway = IPADDR4_INIT_BYTES(0, 0, 0, 0);
+#endif
+
+void ota_server_task(void *pvParameters)
+{
+    char addr_str[128];
```

Comment:
1) The server task should be monitored by the watchdog
2) The watchdog timeout shall cause a reboot